### PR TITLE
Quote submodule path to avoid black screen issue

### PR DESCRIPTION
**Pull request by leorohr**
Quote SubmodulePath in GetSuperProjectCurrentCheckout to handle paths with spaces

Related to #6268 #6243 #6082 

## Proposed changes

- Quote SubmodulePath in GetSuperProjectCurrentCheckout to handle paths with spaces

## Test methodology 

- Manual tests
- Unit tests

## Test environment(s) 

- GIT  2.20.1.windows.1 
- Windows 10 (1709 / 16299.967) 

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by gerhardol** [comment]
Please sign https://github.com/gitextensions/gitextensions/blob/master/contributors.txt first, as a separate PR

**Comment by RussKie** [comment]
Please do a sign off to https://github.com/gitextensions/gitextensions/blob/master/contributors.txt

**Comment by illfated** [comment]
Hah - a "race condition" in comments from the team members. 😄

**Comment by gerhardol** [comment]
> Hah - a "race condition" in comments from the team members.

Will happen, RussKie is a little too far away to here me.

I updated the wiki too, to visualize the signing better

**Comment by RussKie** [comment]
On my morning commute I have a patchy internet connection and things like
this happen...

**Comment by RussKie** [comment]
Could you please resolve the MC

**Comment by RussKie** [comment]
We don't do x-branch merges, please do a clean rebase.

**Comment by leorohr** [comment]
@gerhardol Sorry, didnt see the status change before pushing again. Had to rectify the committer & author (accidentally pushed with my work details).

**Comment by gerhardol** [comment]
merge conflict again - better with separate PR
However, it can be resolved when merging I believe.
I will merge in a day or so, unless someone objects.

**Comment by RussKie** [comment]
I'd like to see tests around it. I'm happy to help with the infra, if necessary.

We can do either a unit test - for that we'll need to mock the executable (please have a look at `GitModuleTest` and `MockExecutable`).
Or we can do an integration test - please have a look at `GitModuleTestHelper` and `ReferenceRepository`.

**Comment by leorohr** [comment]
I'll give the tests a go and update the PR by tomorrow

**Comment by leorohr** [comment]
@RussKie I added a test for the method in question. Is that what you had in mind?

File: UnitTests/GitCommandsTests/GitModuleTest.cs [hunk 1]
```diff
@@ -504,6 +504,41 @@ public void GetSubmodulesLocalPaths()
             }
         }
 
+        [Test]
+        public void GetSuperprojectCurrentCheckout()
+        {
+            // Create super and sub repo
+            var moduleTestHelperSuper = new CommonTestUtils.GitModuleTestHelper("super repo");
+            var moduleTestHelperSub = new CommonTestUtils.GitModuleTestHelper("sub repo");
```
Comment (RussKie) [inline UnitTests/GitCommandsTests/GitModuleTest.cs]: ⚠️ `GitModuleTestHelper` is disposable, please wrap these two in `using`.

Comment (leorohr) [reply UnitTests/GitCommandsTests/GitModuleTest.cs]: Done

**Comment by RussKie** [comment]
Thank you

**Pull request merged**